### ISSUE TITLE
Seeing several branch of a desktop app

### ISSUE THREAD

**Issue by Jehan** [flatpak/flatpak#1109]
So GIMP is now on flathub as "stable" branch, but as you know, I also work with other branches ("dev" for dev releases and "master" for nightlies) and I notice I cannot see them all in GUIs (i.e. GNOME Overview or listed in middle-click menu of GNOME Files, etc.).
Only one branch is visible to the OS (looks like only the last installed branch is).
Since on flatpak.org, "Install different versions of the same application" is one of the 5 main features advertized, I am going to assume that's not normal.

Through command line though, I can still run any of the branches (`flatpak run org.gimp.GIMP//master` for instance).

My assumption is that flatpak renames the desktop file as `org.gimp.GIMP.desktop` for every branch. So the last installed one overrides visibility of previous desktop files. Can't the branch name be used in the renaming? Like `org.gimp.GIMP_stable.desktop`?

**Comment by alexlarsson** [comment]
You can switch which one is active with `flatpak make-current`, but fundamentally, the way desktop integration works (freedesktop specs, dbus, etc) you can only integrate one of them with the desktop at a time.

**Comment by Jehan** [comment]
Well… I'm pretty sure that it's possible! I already see several GIMPs (in particular I see my dev GIMP and one of the flatpak GIMPs, the last installed). What else than the file name is there to have the desktop consider that these are 2 different applications and propose both of them?

This really diminish a lot the interest of saying "install several versions" if you can't run them by GUI. Even I, when not developing, I prefer to run GUI applications through the overview.

**Comment by alexlarsson** [comment]
Well, you can if you change the app id, because then you get a differently named desktop file, icon, dbus service, etc.

**Comment by alexlarsson** [comment]
Thats why my nightly builds were called org.gimp.GimpDevel

**Comment by alexlarsson** [comment]
But you have to be careful to also modify the desktop file and/or the icon so its clear which one is which in the menus.

**Comment by Jehan** [comment]
> Well, you can if you change the app id, because then you get a differently named desktop file, icon, dbus service, etc.

Yes but it's still different conceptually. I really like the concept of having several branches of the same application. Now maybe since the id is not visible to the GUI, it's not that important. Still branches could be nice both while using flatpak in GUI and in terminal.

> But you have to be careful to also modify the desktop file and/or the icon so its clear which one is which in the menus.

Of course. I use "desktop-file-name-prefix" for that, prepending "(dev) " for dev releases and "(nightly) " for master. That makes the 3 versions of GIMP absolutely unmistakeable and works well. :-)
